### PR TITLE
Add log level `OFF`

### PR DESCRIPTION
**Pull request by msievers**
## What does this PR do?

This PR adds a new log level `OFF` to be configured via `elastic.apm.log_level` to silence the apm agent logging.

## Checklist

- [x] My code follows the [style guidelines of this project](CONTRIBUTING.md#java-language-formatting-guidelines)
- [x] I have rebased my changes on top of the latest master branch

- [x] I have made corresponding changes to the documentation
- [ ] ~~I have added tests that prove my fix is effective or that my feature works~~
- [x] New and existing [**unit** tests](https://github.com/elastic/apm-agent-java/blob/master/CONTRIBUTING.md#testing) pass locally with my changes

- [x] I have updated [CHANGELOG.asciidoc](CHANGELOG.asciidoc)
- [ ] ~~I have updated [supported-technologies.asciidoc](docs/supported-technologies.asciidoc)~~
- [ ] Added an API method or config option? Document in which version this will be introduced
- [ ] ~~Added an instrumentation plugin? How did you make sure that old, non-supported versions are not instrumented by accident?~~

## Related issues

* https://github.com/elastic/apm-agent-java/pull/1154#issuecomment-620035271

## Use cases

There are companies that have very strict requirements regarding application logs/log formats which may prevent the usage of Elastic APM agent, because there is currently no way, to configure/alter the log format, for example to output JSON.

As a workaround, it should be possible to silence the APM agent logging.

## Screenshots
<!-- _(Optional)_
Add here screenshots about how the project will be changed after the PR is applied. They could be related to web pages, terminal, etc, or any other image you consider important to be shared with the team.

### CODE REVIEW

**Comment by msievers** [comment]
This PR replaces my former attempt to silence to apm agent logging by making it possible to specify `/dev/null` as the log file (see https://github.com/elastic/apm-agent-java/pull/1154) which would interfere with upcoming changes regarding log file rotation.

File: apm-agent-core/src/main/java/co/elastic/apm/agent/logging/LogLevel.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * #L%
+ */
+package co.elastic.apm.agent.logging;
+
+enum LogLevel {
+    OFF,
```
Comment (felixbarny) [inline apm-agent-core/src/main/java/co/elastic/apm/agent/logging/LogLevel.java:28]: How handy that SimpleLogger already supports `OFF`

Comment (msievers) [reply apm-agent-core/src/main/java/co/elastic/apm/agent/logging/LogLevel.java:28]: Yeah, that makes it quite simple :)

**Comment by eyalkoren** [comment]
Thanks @msievers !!
Just running the CI tests to see that it passes properly before merging

**Pull request merged**